test/Main.test.ts

```diff
@@ -1689,40 +1729,75 @@ describe(`MainP${IMPLEMENTATION} contract`, () => {
     })
 
     it('Should not allow to set prime Basket with 0 address tokens', async () => {
+      await expect(
+        freshBasketHandler.connect(owner).setPrimeBasket([ZERO_ADDRESS], [fp('1')])
+      ).to.be.revertedWith('address zero is not valid collateral')
       await expect(
         basketHandler.connect(owner).setPrimeBasket([ZERO_ADDRESS], [fp('1')])
       ).to.be.revertedWith('address zero is not valid collateral')
     })
 
     it('Should not allow to set prime Basket with stRSR', async () => {
+      await expect(
+        freshBasketHandler.connect(owner).setPrimeBasket([stRSR.address], [fp('1')])
+      ).to.be.revertedWith('stRSR is not valid collateral')
       await expect(
         basketHandler.connect(owner).setPrimeBasket([stRSR.address], [fp('1')])
       ).to.be.revertedWith('stRSR is not valid collateral')
     })
 
-    it('Should not allow to set prime Basket with invalid target amounts', async () => {
+    it('Should not allow to bypass MAX_TARGET_AMT', async () => {
+      // not possible on non-fresh basketHandler
       await expect(
-        basketHandler.connect(owner).setPrimeBasket([token0.address], [MAX_TARGET_AMT.add(1)])
+        freshBasketHandler.connect(owner).setPrimeBasket([token0.address], [MAX_TARGET_AMT.add(1)])
       ).to.be.revertedWith('invalid target amount; too large')
     })
 
+    it('Should not allow to increase prime Basket weights', async () => {
+      // not possible on freshBasketHandler
+      await expect(
+        basketHandler.connect(owner).setPrimeBasket([token0.address], [fp('1').add(1)])
+      ).to.be.revertedWith('new basket adds target weights')
+    })
+
+    it('Should not allow to decrease prime Basket weights', async () => {
+      // not possible on freshBasketHandler
+      await expect(
+        basketHandler.connect(owner).setPrimeBasket([token0.address], [fp('1').sub(1)])
+      ).to.be.revertedWith('new basket missing target weights')
+    })
+
     it('Should not allow to set prime Basket with an empty basket', async () => {
+      await expect(freshBasketHandler.connect(owner).setPrimeBasket([], [])).to.be.revertedWith(
+        'cannot empty basket'
+      )
       await expect(basketHandler.connect(owner).setPrimeBasket([], [])).to.be.revertedWith(
         'cannot empty basket'
       )
     })
 
     it('Should not allow to set prime Basket with a zero amount', async () => {
       await expect(
-        basketHandler.connect(owner).setPrimeBasket([token0.address], [0])
+        freshBasketHandler.connect(owner).setPrimeBasket([token0.address], [0])
       ).to.be.revertedWith('invalid target amount; must be nonzero')
+      await expect(
+        basketHandler.connect(owner).setPrimeBasket([token0.address], [0])
+      ).to.be.revertedWith('new basket missing target weights')
     })
 
     it('Should not allow to set prime Basket with RSR/RToken', async () => {
```

Comment:
Maybe add the case of having all current Target names and amounts equal but adding a new Target name to see how its handled?

Reply:
Yeah I haven't done new tests yet, adding today